esmf_regrid/_esmf_sdo.py

```diff
@@ -3,7 +3,15 @@
 from abc import ABC, abstractmethod
 
 import cartopy.crs as ccrs
-import ESMF
+
+try:
+    import esmpy
+except ImportError as exc:
+    # Prior to v8.4.0, `esmpy`` could be imported as `ESMF`.
+    try:
+        import ESMF as esmpy  # noqa: N811
+    except ImportError:
+        raise exc
```

Comment:
With such an involved series of steps, might it be worth performing this complex import once in the top level `__init__`, then having all other modules just import from there?